KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,327 @@
+    :return: The processed chapter, ready for additional processing
+    """
+    foot_misc_index = 0
     index = 0
     for word in chap:
         index += 1
-        if("footmisc" in word):
-            footmiscIndex += index
-    # edits the chapter string sent to include hyperref, xparse, and cite packages
-    #str[footmiscIndex] += "\\usepackage[pdftex]{hyperref} \n\\usepackage {xparse} \n\\usepackage{cite} \n"
-    chap.insert(
-        footmiscIndex,
-        "\\usepackage[pdftex]{hyperref} \n\\usepackage {xparse} \n\\usepackage{cite} \n")
+        if "footmisc" in word:
+            foot_misc_index += index
+    # str[footmiscIndex] += "\\usepackage[pdftex]{hyperref} \n\\usepackage {xparse} \n\\usepackage{cite} \n"
+    chap.insert(foot_misc_index, "\\usepackage[pdftex]{hyperref} \n\\usepackage {xparse} \n\\usepackage{cite} \n")
     return chap
 
-# 2/18/16 this method reads in relevant commands that are in KLSadd.tex
-# and returns them as a list
 
+def get_commands(kls, new_commands):
+    """
+    This method reads in relevant commands that are in KLS Addendum.tex and returns them as a list
 
-def getCommands(kls):
+    :param kls: The addendum is the source of what is being found
+    :return:
+    """
     index = 0
     for word in kls:
         index += 1
-        if("smallskipamount" in word):
-            newCommands.append(index - 1)
-        if("mybibitem[1]" in word):
-            newCommands.append(index)
-    comms = kls[newCommands[0]:newCommands[1]]
-    return comms
+        if "smallskipamount" in word:
+            new_commands.append(index - 1)
+        if "mybibitem[1]" in word:
+            new_commands.append(index)
+    print new_commands
+    return kls[new_commands[0]:new_commands[1]]
 
-# 2/18/16 this method addresses the goal of hardcoding in the necessary
-# commands to let the chapter files run as pdf's. Currently only works
-# with chapter 9
 
+# 2/18/16 this method addresses the goal of hardcoding in the necessary commands to let the chapter files run as pdf's.
+# Currently only works with chapter 9
+def insert_commands(kls, chap, cms):
+    """
+    Inserts commands identified in previous functions.
 
-def insertCommands(kls, chap, cms):
+    :param kls: Addendum to look through
+    :param chap: The chapter that is receiving commands (9 or 14)
+    :param cms: Commands to be inserted
+    :return:
+    """
     # reads in the newCommands[] and puts them in chap
-    # the index of the "begin document" keyphrase, this is where the new
-    # commands need to be inserted.
-    beginIndex = -1
-    # find index of begin document in KLSadd
+    begin_index = -1  # the index of the "begin document" keyphrase, this is where the new commands need to be inserted.
+    # find index of begin document in KLS Addendum
     index = 0
+
     for word in kls:
         index += 1
-        if("begin{document}" in word):
-            beginIndex += index
+        if "begin{document}" in word:
```

Comment:
considering this is all that is done in this loop, you should probably break there. Also use enumerate here.

It seems that there is a lot of finding of things and storage of indices done. It would be a lot more efficient if you found all of these things once and kept them in a dictionary or something to lookup, then passed the dictionary to all functions that would do this finding. This would be significantly more efficient than looping through the whole list many times.